src/databricks/labs/ucx/contexts/application.py

```diff
@@ -287,15 +287,15 @@ def aws_acl(self):
     @cached_property
     def principal_locations(self):
         eligible_locations = {}
-        if self.is_azure:
-            try:
+        try:
+            if self.is_azure:
                 eligible_locations = self.azure_acl.get_eligible_locations_principals()
-            except ResourceDoesNotExist:
-                pass
-        if self.is_aws:
-            eligible_locations = self.aws_acl.get_eligible_locations_principals()
-        if self.is_gcp:
-            raise NotImplementedError("Not implemented for GCP.")
+            if self.is_aws:
+                eligible_locations = self.aws_acl.get_eligible_locations_principals()
+            if self.is_gcp:
+                raise NotImplementedError("Not implemented for GCP.")
```

Comment:
Is it implemented when all of the conditions are `False` and we return an empty dict?

Reply:
I don't think that can happen, but this is not my code